packages/astro/src/react/controlComponents.tsx

```diff
@@ -70,7 +74,28 @@ export const ClerkLoading = ({ children }: React.PropsWithChildren): JSX.Element
 };
 
 export type ProtectProps = React.PropsWithChildren<
-  ProtectComponentDefaultProps & {
+  (
+    | {
+        condition?: never;
+        role: OrganizationCustomRoleKey;
+        permission?: never;
+      }
+    | {
+        condition?: never;
+        role?: never;
+        permission: OrganizationCustomPermissionKey;
+      }
+    | {
+        condition: (has: CheckAuthorizationWithCustomPermissions) => boolean;
+        role?: never;
+        permission?: never;
+      }
+    | {
+        condition?: never;
+        role?: never;
+        permission?: never;
+      }
+  ) & {
```

Comment:
We need to separate the prop types of the Astro component and React component since the Astro one has no build step and imported types will not be resolvable.

Reply:
Would the astro prop types need `fallback: never;` added?

Reply:
If the purpose of adding `fallback: never` is to show an error whenever unexpected props are used in Astro `<Protect />`, we don't need it it since it should already show type errors for random props

Also just FYI, Astro does not support [typed slots](https://github.com/withastro/roadmap/discussions/414)